Bob crashes if the `--settings` path contains slash

**Describe the bug (REQUIRED)**
The `--settings` option generates an error in bob.jar (the argument is `--settings targets/poki/game.settings`):

```
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Resource 'game.settings' is not created
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.dynamo.bob.TaskBuilder.build(TaskBuilder.java:297)
	at com.dynamo.bob.Project.runTasks(Project.java:1784)
	at com.dynamo.bob.Project.createAndRunTasks(Project.java:1568)
	at com.dynamo.bob.Project.doBuild(Project.java:1707)
	at com.dynamo.bob.Project.build(Project.java:710)
	at com.dynamo.bob.Bob.invoke(Bob.java:978)
	at com.dynamo.bob.Bob.main(Bob.java:1054)
Caused by: java.lang.IllegalArgumentException: Resource 'game.settings' is not created
	at com.dynamo.bob.fs.AbstractResource.sha1(AbstractResource.java:62)
	at com.dynamo.bob.Task.updateDigestWithResources(Task.java:169)
	at com.dynamo.bob.Task.calculateSignatureDigest(Task.java:199)
	at com.dynamo.bob.Task.calculateSignature(Task.java:207)
	at com.dynamo.bob.TaskBuilder.buildTask(TaskBuilder.java:161)
	at com.dynamo.bob.TaskBuilder.lambda$createCallableTask$0(TaskBuilder.java:102)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

If there is a subdirectory in the path in `--settings`, the build crashes with the error above. If the file is in the root, everything works as before.

On 1.9.5 everything works well. 

**Defold version (REQUIRED):**
  - Version 1.9.6, `4035511f1e258a47b77798a2005c024609ed2c67`

**Minimal repro case project (OPTIONAL):**
[issue_196_bob_settings.zip](https://github.com/user-attachments/files/18153061/issue_196_bob_settings.zip)

--- a/com.dynamo.cr/com.dynamo.cr.bob.test/abssettings/absolute.ini
+++ b/com.dynamo.cr/com.dynamo.cr.bob.test/abssettings/absolute.ini
@@ -1,0 +1,2 @@
+[test]
+absolute = 2

--- a/com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/JarTest.java
+++ b/com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/JarTest.java
@@ -98,6 +98,13 @@ public class JarTest {
         assertEquals(1337, result);
     }
 
+    @Test
+    public void testSettings() throws Exception {
+        String[] args = new String[] {"--settings=./test/proj/settings/settings.ini"};
+        int result = bob(args, "");
+        assertEquals(0, result);
+    }
+
     /**
      * The purpose with this test is to be as close as possible to testBuild, but make debugging easier by running the same JVM.
      * @throws Exception

--- a/com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/JarTest.java
+++ b/com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/JarTest.java
@@ -20,6 +20,7 @@ import static org.junit.Assert.assertTrue;
 
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.URISyntaxException;
@@ -36,6 +37,7 @@ import com.dynamo.bob.archive.publisher.NullPublisher;
 import com.dynamo.bob.archive.publisher.PublisherSettings;
 import com.dynamo.bob.fs.DefaultFileSystem;
 import com.dynamo.bob.fs.IFileSystem;
+import com.dynamo.bob.util.BobProjectProperties;
 
 public class JarTest {
 
@@ -100,9 +102,21 @@ public class JarTest {
 
     @Test
     public void testSettings() throws Exception {
-        String[] args = new String[] {"--settings=./test/proj/settings/settings.ini"};
+        File absoluteSettings = new File("abssettings/absolute.ini");
+        String absPath = absoluteSettings.getAbsolutePath();
+
+        String[] args = new String[] {"--settings=./test/proj/settings/settings.ini", "--settings", absPath};
         int result = bob(args, "");
         assertEquals(0, result);
+
+        File projectSettings = new File("test/proj/build/default/game.projectc");
+        assertTrue(projectSettings.exists());
+
+        BobProjectProperties properties = new BobProjectProperties();
+        properties.load(new FileInputStream(projectSettings));
+
+        assertEquals(1, (int)properties.getIntValue("test", "value", 0));
+        assertEquals(2, (int)properties.getIntValue("test", "absolute", 0));
     }
 
     /**

--- a/com.dynamo.cr/com.dynamo.cr.bob.test/test/proj/settings/settings.ini
+++ b/com.dynamo.cr/com.dynamo.cr.bob.test/test/proj/settings/settings.ini
@@ -1,0 +1,2 @@
+[test]
+value = 1

--- a/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java
+++ b/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java
@@ -677,9 +677,13 @@ public class Project {
             if (!path.isAbsolute()) {
                 Path rootDir = Paths.get(getRootDirectory()).normalize().toAbsolutePath();
                 Path settingsFile = path.normalize().toAbsolutePath();
-                path = rootDir.relativize(settingsFile);
+                Path relativePath = rootDir.relativize(settingsFile);
+                resources.add(fileSystem.get(relativePath.toString()));
             }
-            resources.add(fileSystem.get(path.getFileName().toString()));
+            else
+            {
+                resources.add(fileSystem.get(path.getFileName().toString()));
+            }
         }
         return resources;
     }
@@ -728,8 +732,13 @@ public class Project {
      */
     public void mount(IResourceScanner resourceScanner) throws IOException, CompileExceptionError {
         this.fileSystem.clearMountPoints();
-        Set<String> mounts = new HashSet<>();
         this.fileSystem.addMountPoint(new ClassLoaderMountPoint(this.fileSystem, "builtins/**", resourceScanner));
+
+        // This code is a quick way to allow for settings files from outside of a project
+        // Those settings files are required to be using absolute paths.
+        // Caveat is that we're mounting the folder, which may contain other files
+        // TODO: Add way to insert specific (or virtual) files into the file system
+        Set<String> mounts = new HashSet<>();
         for (String propertyFile : propertyFiles) {
             Path path = Paths.get(propertyFile);
             if (path.isAbsolute()) {

--- a/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/TaskBuilder.java
+++ b/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/TaskBuilder.java
@@ -152,16 +152,16 @@ public class TaskBuilder {
         taskResult.setResult(Result.SUCCESS);
         taskResult.setProfilingScope(TimeProfiler.getCurrentScope());
 
-        final List<IResource> outputResources = task.getOutputs();
-        // check if all outputs already exist
-        final boolean allOutputsExist = checkIfResourcesExist(outputResources);
-
-        // compare all task signature. current task signature between previous
-        // signature from state on disk
-        final byte[] taskSignature = task.calculateSignature();
-        final boolean allSigsEquals = compareAllSignatures(taskSignature, outputResources);
-
         try {
+            final List<IResource> outputResources = task.getOutputs();
+            // check if all outputs already exist
+            final boolean allOutputsExist = checkIfResourcesExist(outputResources);
+
+            // compare all task signature. current task signature between previous
+            // signature from state on disk
+            final byte[] taskSignature = task.calculateSignature();
+            final boolean allSigsEquals = compareAllSignatures(taskSignature, outputResources);
+
             Builder builder = task.getBuilder();
             Map<IResource, String> outputResourceToCacheKey = new HashMap<IResource, String>();
             

